### ISSUE TITLE
Add option to not show listbox with no text input

### ISSUE THREAD

**Issue by tarngerine** [i-like-robots/react-tag-autocomplete#15]
Would like the ability to not show the Listbox until you start typing. Thanks!

**Comment by tarngerine** [comment]
minQueryLength feels like a super specific prop, as does closeOnSelect. I wonder if both of those can be enabled by more generic props like: `open`, and `onSelect` (or `onAdd`). This way you can build support for something like minQueryLength yourself by tracking input with `onInput`, and close the element by managing an `open` `useState` and toggling it yourself. you can add an `onOpenChange` prop to allow syncing any internally driven open states that should toggle the controlled open. Radix uses the `open`/`onOpenChange` pattern and it works well https://www.radix-ui.com/docs/primitives/components/dialog

**Comment by i-like-robots** [comment]
The reason I removed `minQueryLength` and haven't implemented an alternative is that with v7 I have attempted to stay as close as possible to both the published [combobox patterns](https://www.w3.org/WAI/ARIA/apg/patterns/combobox/) and observed behaviour of the native datalist component. Neither of these demonstrate any restriction on the listbox expanding.

Implementing an option so implementors may provide a function to control of the listbox would be straightforward by amending [this method](https://github.com/i-like-robots/react-tag-autocomplete/blob/f5e94bc4022d97c475e2a9862b8317618a70550a/src/hooks/useManagerTwo.ts#L123-L125) to something like this:

```ts
    expand() {
      const shouldExpand = typeof onShouldExpand === 'function' ? onShouldExpand(state) : true
      shouldExpand && setIsExpanded(true)
    },
```

Manually expanding or collapsing the listbox is also possible via [the API](https://github.com/i-like-robots/react-tag-autocomplete/blob/f5e94bc4022d97c475e2a9862b8317618a70550a/src/hooks/usePublicAPI.ts#L27-L32) but this is not documented or tested yet.
